### PR TITLE
added start date and end date filter on event page

### PR DESCRIPTION
**Pull request by SushieeK**
added start date and end date filter on event page

### CODE REVIEW

**Comment by SushieeK** [comment]
added start date and end date filter on event page

File: events/views.py [hunk 1]
```diff
@@ -4,13 +4,46 @@
...
+                events = events.filter(start_time__gte=form.cleaned_data["start_time"])
+            if form.cleaned_data["end_time"]:
+                if form.cleaned_data["end_time"] <= timezone.now():
+                    # Return an error message
+                    # Handle this appropriately in your template
+                    return render(
+                        request,
+                        "events/events.html",
+                        {"error": "End time cannot be in the past."},
+                    )
```
Comment (WenjunYuAnny) [inline events/views.py:32]: The filter works great! Thank you! Just one note here: there should be another kind of error message: if the end time filter is earlier than the start time filter, the user should be alerted that there's a mistake in the input. (#61)

File: events/templates/events/events.html [hunk 2]
```diff
@@ -45,6 +45,27 @@
 
 <div class="sticky-note-container-big mt-5">
     <h1>Events</h1>
+
+    <!-- Filtering form -->
+    <form method="get" action="{% url 'events:index' %}" class="filter-form">
+        <label for="start_time">Start Time:</label>
```
Comment (Dhiren-NYU) [inline events/templates/events/events.html:51]: I feel we should add checks that start time cannot be less than current time and end time cannot be less than start time

File: events/views.py [hunk 3]
```diff
@@ -4,13 +4,46 @@
...
+from django.utils import timezone
+from .forms import EventFilterForm
 
 # Create your views here.
 
 
 def index(request):
-    events = Event.objects.filter(is_active=True)
-    return render(request, "events/events.html", {"events": events})
+    events = Event.objects.filter(end_time__gt=timezone.now(), is_active=True).order_by(
```
Comment (Dhiren-NYU) [inline events/views.py:14]: timezone.now() will give u time based on IST
please add Ny time if u want u can refer my pr

File: events/views.py [hunk 4]
```diff
@@ -4,13 +4,46 @@
...
+    )
+
+    # Process the filter form
+    if request.GET:
+        form = EventFilterForm(request.GET)
+        if form.is_valid():
+            if form.cleaned_data["start_time"]:
+                events = events.filter(start_time__gte=form.cleaned_data["start_time"])
+            if form.cleaned_data["end_time"]:
+                if form.cleaned_data["end_time"] <= timezone.now():
```
Comment (Dhiren-NYU) [inline events/views.py:25]: We should add these checks on frontend as well?

**Review by Dhiren-NYU — COMMENTED** [review]
reviewed

File: events/templates/events/events.html (hunk 2 above)
Comment (SushieeK) [reply events/templates/events/events.html:51]: yup, I added the code for it.

File: events/views.py (hunk 1 above)
Comment (SushieeK) [reply events/views.py:47]: added checks for start time cannot be less than current time and end time cannot be less than start time


image

File: events/templates/events/events.html (hunk 2 above)
Comment (SushieeK) [reply events/templates/events/events.html:51]: image
image

File: events/views.py (hunk 3 above)
Comment (SushieeK) [reply events/views.py]: Adressed the concern, used pytz.

File: events/views.py (hunk 4 above)
Comment (SushieeK) [reply events/views.py]: added the checks on frontend.

**Review by Dhiren-NYU — APPROVED** [review]

**Pull request merged**